### DISCUSSION TITLE
IDP Kit use cases

### DISCUSSION THREAD

**Discussion by mvresh** [walt-id/.github#24] [Help] [answered]
Hi,

I'm working on IDP use cases with NFTs - https://docs.walt.id/v/idpkit/tutorials/login-with-nfts-or-next.js and https://docs.walt.id/v/idpkit/tutorials/login-with-nfts-or-keycloak using AWS compute instance, rather than using local machine. 

For the first one, I'm able to deploy the IDP kit on Ethereum mainnet using Alchemy and using the ENS domains NFT collection from Opensea. However, when I deploy the next js app, it says "You are not authenticated" and when I click sign in, its unable to load. 

For the second usecase using Keycloak, I'm unable to get the Keycloak Admin UI panel although keycloak docker is running in the same instance as IDP kit.

Can someone please help resolve this issues. Thanks.

**Comment by taminobaumann** [comment]
Hey @mvresh,

thanks for reaching out and pointing us to this issue!

**First one**
There is currently an issue when pressing the "Sign in with NFT" button on the Next.js app. It returns a not found error and the URL in the browser of the Next.js app is as follows: `http://localhost:8080/connect-wallet?session=122ac335-be03-46b2-81c8-2f8af863ba...` . Please confirm if you are having the same problem!

**Second One**
Here it is important that the local keycloak instance is not running on the same port as the IDP-Kit. This is the case by default and which could explain the problem to not being able to access the admin console of Keycloak. You can fix this issue when starting keycloak as follows: `bin/kc.sh start-dev --http-port 8081`.

**Reply by mvresh** [reply]
**First issue**

Yes. When I `npm run dev` it loads the "Sign in with NFT" on the following URL - `remoteIP:3000/api/auth/signin?callbackUrl=http%3A%2F%2F34.252.211.44%3A3000%2F` normally but when I click on it, it loads `http://localhost/:3000/api/auth/signin/walt-id-nft` which is odd because its supposed to load on the remote IP. I tried editing the URL with remote IP but still its the same issue(can't be reached)

**Second issue**

I installed Keycloak using docker and I'm running it on 8081 but for some reason it still running on 8080 for some reason. Found that the issue could be with the remote Keycloak access. Unable to resolve it yet.

**Comment by taminobaumann** [comment, marked as answer]
@mvresh 

**First issue**

Do you have the environment variable set to the new host in the .env file? See [next-auth docs](https://next-auth.js.org/configuration/options)


**Second Issue**

You can also try to run the IDP-Kit on a different port. This would require a change of the `identityProviderURL` parameter which is passed to the NFTProvider in the next.js app under pages/api/auth/[...nextauth].ts. The value passed would refer to the new host the IDP-Kit is running on.
